doc/developers/contributing.rst

```diff
@@ -356,10 +356,21 @@ complies with the following rules before marking a PR as ``[MRG]``. The
    non-regression tests should fail for the code base in the master branch
    and pass for the PR code.
 
-5. **Make sure that your PR does not add PEP8 violations**. On a Unix-like
-   system, you can run `make flake8-diff`. `flake8 path_to_file`, would work
-   for any system, but please avoid reformatting parts of the file that your
-   pull request doesn't change, as it distracts from code review.
+5. **Make sure that your PR does not add PEP8 violations**. To check the
+   code that you changed, you can run the following command::
+
+        git diff upstream/master -u -- "*.py" | flake8 --diff
+        
+    or `make flake8-diff` which should work on unix-like system.
```

Comment:
If we mention it, let's also change the `Makefile` to use the above command for `flake8-diff`